CONTRIBUTING.md

```diff
@@ -218,443 +205,83 @@ If the feature or fix does not directly impact users, consider using a different
 We test the library code against multiple `python` versions
 and use [`pyenv`](https://github.com/pyenv/pyenv) to manage those. Install `pyenv` by running
 
-```shell
-curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
-```
+following the instruction in [here](https://github.com/pyenv/pyenv?tab=readme-ov-file#getting-pyenv).
 
-To load `pyenv` automatically, add the following lines to your shell's startup script,
-such as `~/.bashrc` or `~/.zshrc`
-(and then either restart the shell, run `exec $SHELL`, or `source` the changed script):
+You would also likely want to setup pyenv-virtualenv to manage multiple environement.
+For more details see: [pyenv-virtualenv](https://github.com/pyenv/pyenv-virtualenv?tab=readme-ov-file#pyenv-virtualenv)
 
-```shell
-export PYENV_ROOT="$HOME/.pyenv"
-export PATH="$HOME/.pyenv/bin:$PATH"
-eval "$(pyenv init --path)"
-eval "$(pyenv virtualenv-init -)"
-```
-
-Then run the following command to set up your environment:
+Once you have everything installed, you can add additional python version, for example python 3.7.17, you could run the
+following commmand:
 
-```shell
-./tools/setup_dev_environment.py
 ```
-
-At the first invocation, this tool will set up multiple python environments, which takes some time.
-You can set up a subset of the target environments to test against, for example:
-
-```shell
-./tools/setup_dev_environment.py --python-versions 3.7 3.8
+pyenv install 3.7.17
 ```
```

Comment:
I think we can skip this part. Contributors mostly shouldn't need to care about different Python versions.

We can recommend using Python 3.7 for development since it's the minimum version we currently support.

Reply:
Just provided as an example, i could remove it